src/main.rs

```diff
@@ -108,7 +79,37 @@ fn main() -> anyhow::Result<()> {
 
     engine.run(&component_args[0])?;
 
-    let _ = hd.join();
+    let _ = async_handle.join();
 
     Ok(())
 }
+
+fn async_tasks(rpc_conf: RpcConfig, otlp_conf: OtlpConfig) -> JoinHandle<Result<()>> {
```

Comment:
The commit message "extract async_tasks to a function" makes me confused.
Maybe you meant "extract async tasks to `async_tasks`"?